ledger/src/snapshot_utils.rs

```diff
@@ -32,7 +34,55 @@ pub const TAR_SNAPSHOTS_DIR: &str = "snapshots";
 pub const TAR_ACCOUNTS_DIR: &str = "accounts";
 pub const TAR_VERSION_FILE: &str = "version";
 
-pub const SNAPSHOT_VERSION: &str = "1.1.0";
+const MAX_SNAPSHOT_DATA_FILE_SIZE: u64 = 32 * 1024 * 1024 * 1024; // 32 GiB
+const VERSION_STRING_V1_1_0: &str = "1.1.0";
+const VERSION_STRING_V1_2_0: &str = "1.2.0";
+const DEFAULT_SNAPSHOT_VERSION: SnapshotVersion = SnapshotVersion::V1_2_0;
+
+type DefaultSerdeContextType = SerdeContextV1_2_0;
```

Comment:
nits: How about `use`-ing `DefaultSerdeContext` in `serde_snapshot.rs` here? I want to avoid these similar definitions if possible. Considering `snapshot_utils` is a consumer of `SerdeContextV1_1_1` and `SerdeContextV1_2_0` from `serde_snapshot`. I don't think it's abstraction leakage for `snapshot_utils` to additionally rely on `serde_snapshot` for the informaton of which version is the default.

Reply:
SerdeContext is a pure implementation detail that in retrospect should never have escaped from serde_snapshot.

DefaultSerdeContext is removed, SerdeContext is renamed TypeContext and SerdeContextV1_1_0 / SerdeContextV1_2_0 are now TypeContextLegacy / TypeContextFuture for the time being.

Once the legacy serialisation style is no longer needed, much of this code can and should be removed, as well as the generic type parameterized implementation if it proves to be more trouble than it's worth (I must admit it is quite awkward, especially in rust).

The only serialisation version specificier exposed to ledger/src/snapshot_utils.rs is the SerdeStyle enum with two members - OLDER and NEWER - which fits in with the intended use here (i.e. there is either one snapshot format or two during upgrade periods, hence an older one and a newer one). The version identifiers "1.1.0" etc. really only matter to snapshot_utils.